### PR TITLE
Changed agility plugin functionality to correctly account for xp on i…

### PR DESCRIPTION
**Pull request by tchar989**
**Notes**
- Changed agility plugin functionality to correctly account for xp on intermediate obstacles. 

  - This resolves a state where leveling up on intermediate obstacles does not recalculate laps left. 

- This fixes the bug outlined in issue [runelite#6317](https://github.com/runelite/runelite/issues/6317).
- Changed List.transform function into JDK native stream function for HotColdClue.java.

**Testing**
- Tested agility changes on my RS account by intentionally getting an agility level through intermediate obstacles. Confirmed that lap count is now updated.

### CODE REVIEW

**Comment by tchar989** [comment]
The cause of the bug was that lapsTillNextLevel in AgilitySession.java was being updated only when experience gained was equal to the lap completion bonus xp: https://tinyurl.com/y4n8k72h. If the amount of xp gained was less than the last obstacle XP, the method would return and lapsTillNextLevel in AgilitySession.java would never be recalculated.

The changes in AgilitySession.java were minor style tweaks. The IncrementLapCount method was actually calculated laps till next level and incrementing lap count, so I separated them out for more clarity.

**Comment by tchar989** [comment]
Looks like 2 checkstyle errors were thrown due to curly bracket placement and an unused import. Working on making those tweaks.

File: runelite-client/src/main/java/net/runelite/client/plugins/agility/AgilitySession.java [hunk 1]
```diff
@@ -45,22 +46,34 @@
...
 
 		int remainingXp;
-		do
+		do 
 		{
 			remainingXp = nextLevel <= Experience.MAX_VIRT_LEVEL ? Experience.getXpForLevel(nextLevel) - currentExp : 0;
 			nextLevel++;
 		} while (remainingXp < 0);
 
 		lapsTillLevel = remainingXp > 0 ? (int) Math.ceil(remainingXp / course.getTotalXp()) : 0;
```
Comment (Adam-) [inline runelite-client/src/main/java/net/runelite/client/plugins/agility/AgilitySession.java]: This value becomes wrong if it is computed at each obstacle in the course

File: runelite-client/src/main/java/net/runelite/client/plugins/cluescrolls/clues/HotColdClue.java [hunk 2]
```diff
@@ -86,7 +87,7 @@ public static HotColdClue forText(String text)
 	@Override
 	public WorldPoint[] getLocations()
 	{
-		return Lists.transform(digLocations, HotColdLocation::getWorldPoint).toArray(new WorldPoint[0]);
+		return digLocations.stream().map(hotColdLocation -> hotColdLocation.getWorldPoint()).collect(Collectors.toList()).toArray(new WorldPoint[0]);
```
Comment (Adam-) [inline runelite-client/src/main/java/net/runelite/client/plugins/cluescrolls/clues/HotColdClue.java]: What?

File: runelite-client/src/main/java/net/runelite/client/plugins/agility/AgilityPlugin.java [hunk 3]
```diff
@@ -195,25 +194,20 @@ public void onExperienceChanged(ExperienceChanged event)
...
-			session.incrementLapCount(client);
-		}
-		else
+		if (session == null || session.getCourse() != course)
 		{
 			session = new AgilitySession(course);
 			// New course found, reset lap count and set new course
 			session.resetLapCount();
-			session.incrementLapCount(client);
+			session.calculateLapsTillLevel(client, skillGained);
```
Comment (Adam-) [inline runelite-client/src/main/java/net/runelite/client/plugins/agility/AgilityPlugin.java]: Why is this being calculated twice?

**Pull request closed without merging**